### PR TITLE
Adapt to LLVM 6.0 changes

### PR DESCRIPTION
**Pull request by vchuravy**
there is now a release_60 branch and this PR supersedes #24845

### CODE REVIEW

File: src/llvm-muladd.cpp [hunk 1]
```diff
@@ -65,15 +65,27 @@ static bool checkCombine(Module *m, Instruction *addOp, Value *maybeMul, Value *
         auto newaddend = builder.CreateFNeg(addend);
         // Might be a const
         if (auto neginst = dyn_cast<Instruction>(newaddend))
+#if JL_LLVM_VERSION >= 60000
```
Comment (yuyichao) [inline src/llvm-muladd.cpp]: These are pre-5.0.

**Review by yuyichao — CHANGES_REQUESTED** [review]

**Review by yuyichao — APPROVED** [review]

**Comment by CodeLenz** [comment]
Hi.

just made a git pull in current master, make cleanall  and set LLVM_VER=6.0.0 in Make.user

> ....
> ....
>     CC src/crc32c.o
>     CC src/processor.o
>     CC src/codegen.o
> /home/lenz/julia/src/codegen.cpp:41:10: fatal error: llvm/Target/TargetSubtargetInfo.h: No such file or directory
>  #include <llvm/Target/TargetSubtargetInfo.h>
>           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> compilation terminated.
> make[1]: *** [Makefile:136: codegen.o] Error 1
> make: *** [Makefile:76: julia-src-release] Error 2
> 

Am I missing some step ? Thanks for your help.

**Comment by vchuravy** [comment]
Can you I unset `LLVM_VER`, then do a `make -C deps uninstall-llvm`, and
then set `LLVM_VER`?

Also note while this fixes compilation on 6.0.0 we still only support LLVM
3.9.1

**Comment by CodeLenz** [comment]
Dear @vchuravy 

No luck with those modifications. 

It seems that  TargetSubtargetInfo.h is now in CodeGen/TargetSubtargetInfo.h instead of Target/TargetSubtargetInfo.h. I tried to change it in codegen.cpp but the error remains. I dont know if some -I/ has to be set in order to make it work. 

Regarding Julia support for newer LLVM versions, I have been using LLVM 5.0.1 along with master for some time, without issues. So, when I saw your patch I and the fact that LLVM 6.0.0 was released ....you know :0)

Thanks!

**Comment by vchuravy** [comment]
Well that probably means I missed and ifelse, will look into it.

**Pull request merged**